snap/info_snap_yaml.go

```diff
@@ -95,6 +96,10 @@ func InfoFromSnapYaml(yamlData []byte) (*Info, error) {
 	if y.Type != "" {
 		typ = y.Type
 	}
+	confinement := ConfinementTypeStrict
+	if y.Confinement != "" {
```

Comment:
Well, the unmarshaler isn't actually called if the given field is missing-- right?